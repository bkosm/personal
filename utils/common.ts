import { replace } from "lodash";

export const getExtension = (path: string) => path.split(".").at(-1)!;
export const getPrefix = (path: string) => path.split(".").at(0)!;
export const formatPostName = (raw: string) => replace(raw, /[_]/g, " ");

export function getInProd<T>(element: T): T | undefined {
  return (Deno.env.get("IS_PROD") === "true" && element) || undefined;
}

export function getStaticUrl(path: string): string {
  const root = Deno.env.get("STATIC_DIR_ROOT");

  if (!root) {
    throw new Error("STATIC_DIR_ROOT is not set");
  }

  return `${root}${path}`;
}

export type PostMeta = {
  lastUpdate: Date;
  creationDate: Date;
  title: string;
};

export async function loadPosts(
  path: string,
): Promise<{ [id: string]: PostMeta }> {
  let posts = {};

  for await (const f of Deno.readDir(path)) {
    if (getExtension(f.name) === "md") {
      const postId = f.name.split(".")[0];
      const meta = JSON.parse(
        await Deno.readTextFile(`${path}/${postId}.json`),
      );
      posts = {
        ...posts,
        [postId]: {
          ...meta,
          lastUpdate: new Date(meta.lastUpdate),
          creationDate: new Date(meta.creationDate),
        } as PostMeta,
      };
    }
  }

  return posts;
}
