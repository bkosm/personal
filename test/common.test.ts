import { loadPosts } from "../utils/common.ts";
import { assertSnapshot, name } from "./deps.ts";

Deno.test("loadPosts", async (t) => {
  const res = await loadPosts("./test/__resources__/posts");

  await t.step("works", async () => {
    await assertSnapshot(t, res, name("test posts"));
  });
});
