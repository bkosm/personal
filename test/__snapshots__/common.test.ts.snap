export const snapshot = {};

snapshot[`test posts 1`] = `
{
  another: {
    creationDate: 2078-08-13T12:25:07.168Z,
    lastUpdate: 2078-08-13T15:12:09.307Z,
    title: "Another woah",
  },
  test: {
    creationDate: 2077-08-13T12:25:07.168Z,
    lastUpdate: 2077-08-13T15:12:09.307Z,
    title: "Woah",
  },
}
`;
